nexus/db-queries/src/db/datastore/deployment.rs

```diff
@@ -641,6 +643,88 @@ impl DataStore {
         Ok(())
     }
 
+    /// Set the current target blueprint's `enabled` field
+    ///
+    /// In order to change the enabled field, `target` must already be the
+    /// current target blueprint
+    pub async fn blueprint_target_set_current_enabled(
```

Comment:
I feel like it's worth a comment explaining why this needs to be different from the other code path for setting the target.  I assume the reason is that that code path would fail because it verifies that the parent blueprint is the current target?  Assuming that's the reason, did you consider having that succeed if the _current_ blueprint matches the one we're trying to set?

Reply:
Hmm, I thought I already addressed that (if briefly) in the doc comments stating the requirements of the `target`. I can expand those, but maybe you're asking for something different?

I did consider just funneling this through `blueprint_target_set_current` and changing its requirements, and opted not to for two reasons:

* These really are different operations, so I think it makes sense for them to have error conditions that are specific to each. In particular, I think it's useful to have a "you tried to set a new target blueprint but you can't because it already is the target"  error possibility. Maybe that's wrong and we'd like it to be more idempotent? I am not sure about this.
* The `InsertTargetQuery` CTE is kind of hairy already. Changing its checks to allow the current blueprint matching in addition to the parent is certainly doable, but would be more work than making them separate.

Reply:
That makes sense, although:

> These really are different operations, so I think it makes sense for them to have error conditions that are specific to each. In particular, I think it's useful to have a "you tried to set a new target blueprint but you can't because it already is the target" error possibility. Maybe that's wrong and we'd like it to be more idempotent? I am not sure about this.

I do think it'd be a little better if it were idempotent.  I raised this because as a caller it seems like it could be a little annoying that you can't just construct a `BlueprintTarget` matching whatever you (or the user) currently wants.  You have to invoke a different function (or not invoke it) based on what changed about the struct.  But given your other point about the hairy CTE (which I totally get), I don't think it makes sense to change any of this now.  We can always revisit if/when we make this part of the external API -- we're not committing to much here.

The doc comment explains the restriction, but not really why.  I'm just thinking something like:

```rust
// Although this function is like `blueprint_target_set_current()` in that both store the given `BlueprintTarget` into the table, the functions are distinct because the preconditions and error cases are different. 
```

(and it's not a big deal either way -- it was just a small point of confusion for me reading the code)